src/sota_tools/ostree_object.h

```diff
@@ -111,6 +111,7 @@ class OSTreeObject {
   std::stringstream http_response_;
   CURL* curl_handle_;
   struct curl_httppost* form_post_;
+  curl_slist* headers;
```

Comment:
We should initialize this to nullptr. Otherwise, it could be anything and will segfault in destructor. This should fix some of the failing tests in CI, I think.